### DISCUSSION TITLE
Артикул

### DISCUSSION THREAD

**Discussion by gochicus** [torin-asakura/masterok#5] [Ideas]
[doc link](https://docs.yandex.ru/docs/view?url=ya-disk-public%3A%2F%2Fb40QzLK90%2Feie4pk6KFOOC7c4vsQRtL0czkby%2Ff%2BtcY%2Fi1vo0lwm1%2Ff02LNhb1HPG7pWpVQYsubyrcajHimPFw%3D%3D&name=%D0%94%D0%BE%D0%BA%D1%83%D0%BC%D0%B5%D0%BD%D1%82%D0%B0%D1%86%D0%B8%D1%8F%20API%20%D0%B4%D0%BB%D1%8F%20%D0%BA%D0%BB%D0%B8%D0%B5%D0%BD%D1%82%D0%BE%D0%B2.docx)
Предварительно, буквенно-цифровое обозначение представляется мне так: E-RS-EG20-IEK-EC42-N-V-SKL
 **E** -  категория, в нашем случае электроника, данное обозначение применяется с расчетом на будущее, исходим из того, что артикулы будут создаваться для разных категорий товаров.
**RS** - двухсимвольное сокращение, обозначающее поставщика.
**EG20** - буквенно-цифровое обозначение для подгруппы товаров , в API характеристики у каждой группы, обозначенной номером есть соответствие в виде поле ETEM_GROUP_NAME обозначающее имя подкатегории.  Мне представляется, что данное буквенно-цифровое обозначение будет аналогично поиску по slug, который мы применяли на GQL, когда делали библиотеку. 
**IEK** - трехбуквенное обозначение брэнда товара.
**EC42** - буквенно-цифровое обозначение, соответствующее полю ETIM_CLASS в API  характеристики. 
**N** - буквенное обозначение поля NAME объекта specs в API  Характеристики. Думаю над стандартизацией обозначений, поскольку  товары очень разные.
**V** - цифровое обозначение поля VVALUE объекта specs в API  Характеристики.
**SKL** - категория позиции , аналогичная CATEGORY из API список позиций.

**Comment by TorinAsakura** [comment]
> [doc link](https://docs.yandex.ru/docs/view?url=ya-disk-public%3A%2F%2Fb40QzLK90%2Feie4pk6KFOOC7c4vsQRtL0czkby%2Ff%2BtcY%2Fi1vo0lwm1%2Ff02LNhb1HPG7pWpVQYsubyrcajHimPFw%3D%3D&name=%D0%94%D0%BE%D0%BA%D1%83%D0%BC%D0%B5%D0%BD%D1%82%D0%B0%D1%86%D0%B8%D1%8F%20API%20%D0%B4%D0%BB%D1%8F%20%D0%BA%D0%BB%D0%B8%D0%B5%D0%BD%D1%82%D0%BE%D0%B2.docx)
> Предварительно, буквенно-цифровое обозначение представляется мне так: E-RS-EG20-IEK-EC42-N-V-SKL
> **E** - категория, в нашем случае электроника, данное обозначение применяется с расчетом на будущее, исходим из того, что артикулы будут создаваться для разных категорий товаров.
> **RS** - двухсимвольное сокращение, обозначающее поставщика.
> **EG20** - буквенно-цифровое обозначение для подгруппы товаров , в API характеристики у каждой группы, обозначенной номером есть соответствие в виде поле ETEM_GROUP_NAME обозначающее имя подкатегории. Мне представляется, что данное буквенно-цифровое обозначение будет аналогично поиску по slug, который мы применяли на GQL, когда делали библиотеку.
> **IEK** - трехбуквенное обозначение брэнда товара.
> **EC42** - буквенно-цифровое обозначение, соответствующее полю ETIM_CLASS в API характеристики.
> **N** - буквенное обозначение поля NAME объекта specs в API Характеристики. Думаю над стандартизацией обозначений, поскольку товары очень разные.
> **V** - цифровое обозначение поля VVALUE объекта specs в API Характеристики.
> **SKL** - категория позиции , аналогичная CATEGORY из API список позиций.

Как по мне - начинается оверхед. Артикул не средство общения, не носитель информации, артикул нужен для быстрой идентификации.
- Думаю, что поставщика можно упразднить, иначе очень много вариаций будет.
- Подгруппу нет смысла ставить, она как и поставщик - вариабельна.
- ETIM вещь доступная к пониманию только тем, кто работает "внутри", для обычных сотрудников эта вещь недоступна, нет смысла её выносить
